src/async_impl/decoder.rs

```diff
@@ -184,6 +210,38 @@ impl Decoder {
         is_brotli
     }
 
+    #[cfg(feature = "deflate")]
+    fn detect_deflate(headers: &mut HeaderMap) -> bool {
```

Comment:
I realize this method was already duplicated, but now seeing a 3rd variant, it seems like a good candidate to change to `fn detect_encoding(headers, encoding_str)`. Not a requirement for merging, but if you felt up to it :)